hyperspy/drawing/signal1d.py

```diff
@@ -484,28 +491,32 @@ def update(self, force_replot=False, render_figure=True,
         else:
             self.line.set_ydata(ydata)
 
-        if 'x' in self.autoscale:
+        # Don't change xlim if axis has 0 length (unnecessary)
+        if 'x' in self.autoscale and len(axis) > 0:
             self.ax.set_xlim(axis[0], axis[-1])
 
-        if 'v' in self.autoscale:
+        # Don't change ymin if data has 0 length (unnecessary)
+        if 'v' in self.autoscale and len(ydata) > 0:
             self.ax.relim()
             y1, y2 = np.searchsorted(axis, self.ax.get_xbound())
             y2 += 2
             y1, y2 = np.clip((y1, y2), 0, len(ydata - 1))
             clipped_ydata = ydata[y1:y2]
-            with ignore_warning(category=RuntimeWarning):
-                # In case of "All-NaN slices"
-                y_max, y_min = (np.nanmax(clipped_ydata),
-                                np.nanmin(clipped_ydata))
-
-            if self._plot_imag:
-                # Add real plot
-                yreal = self._get_data(real_part=True)
-                clipped_yreal = yreal[y1:y2]
+            if len(clipped_ydata) > 0:
                 with ignore_warning(category=RuntimeWarning):
                     # In case of "All-NaN slices"
-                    y_min = min(y_min, np.nanmin(clipped_yreal))
-                    y_max = max(y_max, np.nanmin(clipped_yreal))
+                    y_max, y_min = (np.nanmax(clipped_ydata),
+                                    np.nanmin(clipped_ydata))
+
+                if self._plot_imag:
+                    # Add real plot
+                    yreal = self._get_data(real_part=True)
+                    clipped_yreal = yreal[y1:y2]
+                    with ignore_warning(category=RuntimeWarning):
+                        # In case of "All-NaN slices"
+                        y_min = min(y_min, np.nanmin(clipped_yreal))
+                        y_max = max(y_max, np.nanmin(clipped_yreal))
+
             if y_min == y_max:
```

Comment:
With the addition of the conditional execution of code above, it can happen that the execution reaches this line without having defined ``y_min`` and ``y_max``.